crates/starknet/src/messaging/mod.rs

```diff
@@ -0,0 +1,189 @@
+        ))
+    }
+
+    /// Collects all messages found between
+    /// `from` to the Latest Starknet block, including both blocks.
+    ///
+    /// # Arguments
+    /// * `from` - The block id from which (and including which) the messages are collected.
+    pub async fn collect_messages_to_l1(&self, from: u64) -> DevnetResult<Vec<MsgToL1>> {
+        match self.blocks.get_blocks(Some(BlockId::Number(from)), None) {
+            Ok(blocks) => {
+                let mut messages = vec![];
+
+                for block in blocks {
+                    messages.extend(self.get_block_messages(block)?);
+                }
+
+                Ok(messages)
+            }
+            Err(e) => {
+                if let Error::NoBlock = e {
+                    // We're 1 block ahead of latest block, no messages can be collected.
+                    Ok(vec![])
+                } else {
+                    Err(e)
+                }
+            }
+        }
+    }
+
+    /// Collects and sends to L1 all messages found between
+    /// `from` to the Latest Starknet block, including both blocks.
+    /// Returns the list of messages that were collected and the last processed block.
+    ///
+    /// # Arguments
+    /// * `from` - The block id from which (and including which) the messages are collected.
+    pub async fn collect_and_send_messages_to_l1(
+        &self,
+        from: u64,
+    ) -> DevnetResult<(Vec<MsgToL1>, u64)> {
+        let messaging = if let Some(m) = &self.messaging {
+            m
+        } else {
+            return Err(Error::MessagingError(MessagingError::NotConfigured));
+        };
+
+        match self.blocks.get_blocks(Some(BlockId::Number(from)), None) {
+            Ok(blocks) => {
+                let mut messages = vec![];
+
+                let mut last_processed_block: u64 = 0;
+                for block in blocks {
+                    messages.extend(self.get_block_messages(block)?);
+
+                    last_processed_block = block.header.block_number.0;
+                }
+
+                messaging.send_mock_messages(&messages).await?;
+
+                Ok((messages, last_processed_block))
+            }
+            Err(e) => {
+                if let Error::NoBlock = e {
+                    // We're 1 block ahead of latest block, no messages can be collected.
+                    Ok((vec![], from))
+                } else {
+                    Err(e)
+                }
+            }
+        }
+    }
+
+    /// Fetches all messages from L1 and executes them by executing a `L1HandlerTransaction`
+    /// for each one of them.
+    pub async fn fetch_and_execute_messages_to_l2(
+        &mut self,
+    ) -> DevnetResult<Vec<L1HandlerTransaction>> {
+        if self.messaging.is_none() {
+            return Err(Error::MessagingError(MessagingError::NotConfigured));
+        }
+
+        let chain_id = self.chain_id().to_felt();
+        let messaging = self.messaging.as_mut().unwrap();
```

Comment:
Good point, [I added a getter](https://github.com/glihm/starknet-devnet-rs/blob/7dd5d6e0fe8f14aca40ebd562cc3c5d5e16ce873/crates/starknet/src/messaging/mod.rs#L62) to ease the retrieval of this value, with a `Result` instead of an `expect`.